### PR TITLE
[XLA] Add a 'custom' fusion op type

### PR DESCRIPTION
**Pull request by DavidNorman**
This allows backends that may have fusions that are not included in the main enumeration.

### CODE REVIEW

**Comment by martinwicke** [comment]
@eliben are you reviewing this?

File: tensorflow/compiler/xla/service/hlo_instruction.h [hunk 1]
```diff
@@ -62,6 +62,8 @@ class HloInstruction {
     kTransposeDot,        // Fused into a dot with transposed operands.
     kConvBackwardFilter,  // Fused into a backward filter convolution.
     kConvBackwardInput,   // Fused into a backward input convolution.
+
+    kCustom,              // Custom fusions.
```
Comment (eliben) [inline tensorflow/compiler/xla/service/hlo_instruction.h]: Can you elaborate a bit more in this comment, mentioning backend-specific fustions

**Review by eliben — CHANGES_REQUESTED** [review]

**Review by eliben — APPROVED** [review]

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by DavidNorman** [comment]
cool - thanks

**Comment by DavidNorman** [comment]
@martinwicke Hi there.  Could this be merged now?  Cheers

**Pull request merged**